### PR TITLE
Virtualenv compat

### PR DESCRIPTION
**Pull request by jstucke**
This PR should (hopefully) make FACT compatible to running Python from a venv.

- converted python package dependencies to pip dependencies 
- updated `tlsh` function for newer version 
    - added a test
    - bumped hash/tlsh plugin versions
    - observation: newer tlsh version is backwards compatible to old hashes and can compute hashes for string lenghts >=50 (old: >=256)
        - old tlsh version is not upwards compatible with new hashes (they have a new prefix)
- refactoring

### CODE REVIEW

**Comment by dorpvom** [comment]
Fedora build breaks on pip installation of ssdeep

File: src/plugins/analysis/hash/code/hash.py [hunk 1]
```diff
@@ -34,11 +34,11 @@ def process_object(self, file_object):
...
         file_object.processed_analysis[self.NAME] = {}
-        for h in self.hashes_to_create:
-            if h in algorithms_available:
-                file_object.processed_analysis[self.NAME][h] = get_hash(h, file_object.binary)
+        for hash_ in self.hashes_to_create:
+            if hash_ in algorithms_available:
+                file_object.processed_analysis[self.NAME][hash_] = get_hash(hash_, file_object.binary)
             else:
-                logging.debug('algorithm {} not available'.format(h))
+                logging.debug('algorithm {} not available'.format(hash_))
```
Comment (dorpvom) [inline src/plugins/analysis/hash/code/hash.py:41]: tzz, now here was an obvious refactoring-to-fstring-possibility

File: src/install/common.py [hunk 2]
```diff
@@ -46,28 +47,25 @@ def main(distribution):  # pylint: disable=too-many-statements
...
-        dnf_install_packages('python3-ssdeep')
     else:
         apt_install_packages('libmagic-dev')
         apt_install_packages('libfuzzy-dev')
-        apt_install_packages('python3-tlsh')
-        pip3_install_packages('ssdeep')
 
     pip3_install_packages('git+https://github.com/fkie-cad/fact_helper_file.git')
-    pip3_install_packages('psutil', 'pytest', 'pytest-cov', 'pylint', 'python-magic', 'xmltodict', 'yara-python==3.7.0', 'appdirs', 'flaky')
+    pip3_install_packages('psutil', 'pytest', 'pytest-cov', 'pylint', 'python-magic', 'xmltodict', 'yara-python==3.7.0', 'appdirs', 'flaky', 'python-tlsh', 'ssdeep')
```
Comment (dorpvom) [inline src/install/common.py:68]: No wonder this breaks on fedora. Why was ssdeep moved out of the conditional block?

File: src/helperFunctions/install.py [hunk 3]
```diff
@@ -239,3 +240,7 @@ def load_main_config() -> configparser.ConfigParser:
         raise InstallationError('Could not load config at path {}'.format(config_path))
     config.read(str(config_path))
     return config
+
+
+def is_virtualenv() -> bool:
```
Comment (dorpvom) [inline src/helperFunctions/install.py:245]: Uargh. Short explanation? (StackOverflow post this was copied from suffices)

**Review by dorpvom — CHANGES_REQUESTED** [review]

File: src/helperFunctions/install.py (hunk 3 above)
Comment (jstucke) [reply src/helperFunctions/install.py:245]: https://stackoverflow.com/questions/1871549/determine-if-python-is-running-inside-virtualenv :sweat_smile:

File: src/install/common.py (hunk 2 above)
Comment (jstucke) [reply src/install/common.py:68]: I was under the impression that it worked in  #618. Surely, I have been mistaken. -> back to the conditional block

File: src/plugins/analysis/hash/code/hash.py (hunk 1 above)
Comment (jstucke) [reply src/plugins/analysis/hash/code/hash.py:41]: changed

**Review by dorpvom — APPROVED** [review]

**Pull request merged**